glib/src/gobject/interface_info.rs

```diff
@@ -0,0 +1,16 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+wrapper! {
+    #[derive(Debug)]
+    #[doc(alias = "GInterfaceInfo")]
+    pub struct InterfaceInfo(BoxedInline<gobject_ffi::GInterfaceInfo>);
```

Comment:
As this has no `GType` assigned and also otherwise no memory management functions, maybe this should be implemented fully manually? There's no point in ever cloning these, for example, is there? So it could just be a `&` reference returned from the functions

Reply:
(same for the other two types here)

Reply:
You'd define a

```rust
#[repr(transparent)]
pub struct InterfaceInfo(gobject_ffi::GInterfaceInfo);
```

and then can do things like

```rust
let ptr: *const gobject_ffi::GInterfaceInfo;
let info: &InterfaceInfo = &*(ptr as *const InterfaceInfo);
```

As there's no memory management associated with these structs in C, inventing that on the bindings side is going to be problematic.

Reply:
You can't have `ToGlibPtr::to_glib_full()` for these and also not `FromGlibPtrNone` and `IntoGlib` and `FromGlib`. Those implementations conceptually make no sense.

For the time being you will have to implement the functions with these types manually to get the correct behaviour (borrowing).

Reply:
This is very basic infrastructure so it's quite expected that gir won't do the right thing here and manual bindings are needed in many places.